File: algorithm/leetcode14_longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
    	if(strs.empty())
    		return "";
        string ans = strs[0];
        int i, j;
        for (i = 1; i < strs.size(); ++i)
        {
        	for(j = 0; j < min(strs[i].size(),ans.size()); ++j)
        	{
        		if(strs[i][j] != ans[j])
        			break;
        	}
        	ans = ans.substr(0,j);
        	if(ans.size() == 0)
        		break;
        }
        return ans;
    }
};

File: algorithm/Python3_version/leetcode14.longest-common-prefix.py
class Solution:
	def longestCommonPrefix(self, strs: List[str]) -> str:
		if not strs:
			return ""
		s1 = min(strs)
		s2 = max(strs)
		n = min(len(s1), len(s2))
		for i in range(n):
			if s1[i] != s2[i]:
				return s1[:i]
		return s1